### PR TITLE
Fixes #153: Merge timezones use cases

### LINKED ISSUE #153: Merge All timezones and Main timezone use cases
**Is your feature request related to a problem? Please describe.**
Every timezone (main and others) should be evaluated at the same time, to ease the extraction process. 

**Describe the solution you'd like**
Modify the All Timezones use case. GTFS Metadata "all_timezones" attribute should be modified to "other_timezones", excluding  "main_timezone". Update the creation of dataset entity afterward.

### PR DESCRIPTION
**Pull request by maximearmstrong**
**Summary:**

Fixes #153: Merge timezones use cases so that main timezone and other timezones are separated for the dataset entity creation. 

The main timezone should be set in a `PREFERED` rank, but any other timezones should be set in a `NORMAL` rank.

Changes to:

- `process_all_timezones_for_gtfs_metadata.py` is now `process_timezones_for_gtfs_metadata.py` and processes both main and other timezones in one use case as both are dependent.
- `main.py` has been modified to reflect changes
- tests have been modified to reflect changes
- `create_dataset_entity_for_gtfs_metadata.py` is now adding `metadata.other_timezones` to the dataset entity.

**Expected behavior:** 

`main_timezone` and `other_timezones` are now extracted by `process_timezones_for_gtfs_metadata.py`. `create_dataset_entity_for_gtfs_metadata.py` adds the `other_timezones` to the dataset entity.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `pytest` to make sure you didn't break anything
- [x] Format the title like "<short description of fix and changes>" (for example - "Check for null value before using field")
- [x] Linked all relevant issues
- [ ] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: usecase/process_timezones_for_gtfs_metadata.py [hunk 1]
```diff
@@ -15,23 +15,28 @@ def process_all_timezones_for_gtfs_metadata(gtfs_representation):
     dataset = gtfs_representation.dataset
     metadata = gtfs_representation.metadata
 
+    # Extract main timezone
+    main_timezone = dataset.agency[AGENCY_TIMEZONE_KEY].iloc[AGENCY_TIMEZONE_IDX]
+
     # Extract the all the timezones using the stop_timezone in the dataset stops
```
Comment (f8full) [inline usecase/process_timezones_for_gtfs_metadata.py:21]: the all? ;)

File: usecase/test/test_process_timezones_for_gtfs_metadata.py [hunk 2]
```diff
@@ -14,25 +14,25 @@
 TORONTO_TIMEZONE = "America/Toronto"
 
 
-class TestProcessMainTimezoneForGtfsMetadata(TestCase):
-    def test_process_all_timezones_with_none_gtfs_representation_should_raise_exception(
+class TestProcessTimezonesForGtfsMetadata(TestCase):
+    def test_process_timezones_with_none_gtfs_representation_should_raise_exception(
```
Comment (annaelleduff) [inline usecase/test/test_process_timezones_for_gtfs_metadata.py:18]: you can get rid of should_...

File: usecase/process_timezones_for_gtfs_metadata.py (hunk 1 above)
Comment (maximearmstrong) [reply usecase/process_timezones_for_gtfs_metadata.py]: Good point, thanks!

File: usecase/test/test_process_timezones_for_gtfs_metadata.py (hunk 2 above)
Comment (maximearmstrong) [reply usecase/test/test_process_timezones_for_gtfs_metadata.py]: Done in 809bfd0

**Review by annaelleduff — APPROVED** [review]

**Pull request merged**